qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/IndexingIT.java

```diff
@@ -410,19 +409,22 @@ private void assertCount(String index, int count) throws IOException {
         );
     }
 
-    private Version minNodeVersion() throws IOException {
+    private boolean nodeSupportBulkApi(Map<?, ?> nodeInfo) {
+        // TODO[lor]: replace this check with a (historical) feature check ("supports bulk requests")
+        var versionString = nodeInfo.get("version").toString();
+        var version = Version.fromString(versionString);
+        return version.onOrAfter(Version.V_7_5_0);
+    }
+
+    private boolean allNodesSupportBulkApi() throws IOException {
         Map<?, ?> response = entityAsMap(client().performRequest(new Request("GET", "_nodes")));
         Map<?, ?> nodes = (Map<?, ?>) response.get("nodes");
-        Version minNodeVersion = null;
+
+        boolean allSupportBulkApi = true;
         for (Map.Entry<?, ?> node : nodes.entrySet()) {
             Map<?, ?> nodeInfo = (Map<?, ?>) node.getValue();
-            Version nodeVersion = Version.fromString(nodeInfo.get("version").toString());
-            if (minNodeVersion == null) {
-                minNodeVersion = nodeVersion;
-            } else if (nodeVersion.before(minNodeVersion)) {
-                minNodeVersion = nodeVersion;
-            }
+            allSupportBulkApi &= nodeSupportBulkApi(nodeInfo);
         }
-        return minNodeVersion;
+        return allSupportBulkApi;
```

Comment:
I like an in the wild use of `&=`, but maybe this is more concise? We could at least iterate over values and not the entryset.

```
return nodes.values().stream()
            .map(o -> (Map<?, ?>) o)
            .allMatch(this::nodeSupportBulkApi);
```

Reply:
Nicer! Will change it, thanks